### PR TITLE
#3582 delete ruleset

### LINKED ISSUE #3582: [RulesDashboard] - Delete Ruleset
### Description

We need to be able to delete a Ruleset from the in the backend. This should be under the route `rules/ruleset/:rulesetId/delete` We want to soft delete the ruleset by setting its deletedBy field rather than actually deleting. Only admins should be able to delete rulesets.

### Acceptance Criteria

- [ ] deleteRuleset route, controller, and service method
- [ ] only admin can delete ruleset
- [ ] Ensure that edge cases (ruleset doesnt exist or already deleted) are handled
- [ ] unit tests added
- [ ] screenshot of manual tests from postman

### Proposed Solution

Add the deleteRuleset route, controller, and service method. Add unit tests to ensure deletion works properly, and shows errors thrown when ruleset don't exist or were already deleted, or users don't have proper permissions. 

Plenty of examples of delete in the backend, look at the delete partTag or delete part in the parts route/controller/service.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by chrisdicono**
## Changes

Created the delete ruleset endpoint.

## Test Cases

- Ruleset can only be deleted by admins
- Rulesets that don't exist cannot be deleted
- Admins can successfully delete rulesets that do exist, and the deleted ruleset is returned
- Rulesets that have already been deleted cannot be deleted again

## Screenshots

Screenshot 2025-10-19 205738
Screenshot 2025-10-19 205810
Screenshot 2025-10-19 205906
Below are screenshots of the full JSON that was outputted:
Screenshot 2025-10-19 210035
Screenshot 2025-10-19 210053
Screenshot 2025-10-19 210117
Screenshot 2025-10-19 210129
Screenshot 2025-10-19 210234

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #3582

### CODE REVIEW

File: src/shared/src/types/rules-types.ts [hunk 1]
```diff
@@ -30,6 +30,8 @@ export interface Ruleset {
     carId: string;
     name: string;
   };
+  deletedBy: User | null;
```
Comment (cielbellerose) [inline src/shared/src/types/rules-types.ts]: frontend doesnt need to see deleted rulesets since we keep track of soft-deleted rulesets in the backend prisma type

File: src/backend/src/transformers/rules.transformer.ts [hunk 2]
```diff
@@ -11,3 +14,35 @@ export const projectRuleTransformer = (projectRule: any): ProjectRule => {
...
+        name: ruleset.name
+      },
+      subRuleIds: [],
+      referencedRules: [],
+      referencedBy: [],
+      projects: []
+    })),
+    rulesetType: {
+      ...ruleset.rulesetType,
+      lastUpdated: new Date(),
```
Comment (cielbellerose) [inline src/backend/src/transformers/rules.transformer.ts]: we should grab this from the ruleset types lastUpdated field instead since prisma already keeps track of the update time automatically

File: src/backend/src/transformers/rules.transformer.ts [hunk 3]
```diff
@@ -11,3 +14,35 @@ export const projectRuleTransformer = (projectRule: any): ProjectRule => {
...
 };
+
+export const rulesetTransformer = (ruleset: Prisma.RulesetGetPayload<RulesetQueryArgs>): Ruleset => {
+  return {
+    fileId: ruleset.fileId,
+    rulesetId: ruleset.rulesetId,
+    name: ruleset.name,
+    dateCreated: ruleset.dateCreated,
+    active: ruleset.active,
+    rules: ruleset.rules.map((rule) => ({
```
Comment (cielbellerose) [inline src/backend/src/transformers/rules.transformer.ts]: our mistake but do you think you could make this just an assignedPercentage instead of a list of rules? it should calculate what percentage of the ruleset's rules have at least one team. We can update the shared type to be a number for the percentage instead of the "rule: Rule[]" field. This is since the ruleset view frontend doesn't need to display all the rules until we click the view rule button. lmk if you have questions about this!

File: src/backend/src/prisma-query-args/rules.query-args.ts [hunk 4]
```diff
@@ -32,3 +33,16 @@ export const getProjectRuleQueryArgs = () =>
       }
     }
   });
+
+export type RulesetQueryArgs = ReturnType<typeof getRulesetQueryArgs>;
+
+export const getRulesetQueryArgs = (organizationId: string) =>
+  Prisma.validator<Prisma.RulesetDefaultArgs>()({
+    include: {
+      rules: { where: { dateDeleted: null } },
```
Comment (cielbellerose) [inline src/backend/src/prisma-query-args/rules.query-args.ts]: could we have this only keep track of how many teams each rule has instead of querying the full rules

**Review by cielbellerose — CHANGES_REQUESTED** [review]
great work! just a few shared type and query arg comments but this looks fantastic

File: src/shared/src/types/rules-types.ts [hunk 5]
```diff
@@ -25,7 +25,7 @@ export interface Ruleset {
   dateCreated: Date;
   active: boolean;
   rulesetType: RulesetType;
-  rules: Rule[];
+  rules: number;
```
Comment (cielbellerose) [inline src/shared/src/types/rules-types.ts]: could you change the naming to be assignedPercentage here? otherwise everything looks great to me

**Review by cielbellerose — APPROVED** [review]
great work!!!
